### PR TITLE
Support using tags for namespacing keys in session length middleware

### PR DESCRIPTION
**Pull request by justinvdm**
Second attempt at #922. The plan is to have the session length middleware optionally use tags for namespacing redis keys instead of the transport name.

### CODE REVIEW

**Comment by justinvdm** [comment]
Ready for review. I've got a feeling I'm forgetting something though.

File: vumi/middleware/tests/test_session_length.py [hunk 1]
```diff
@@ -136,47 +148,224 @@ def test_incoming_message_session_none_no_overwrite(self):
...
+    def test_incoming_message_session_start_no_transport_name(self):
+        mw = yield self.mk_middleware(namespace_type='transport_name')
+        msg = self.mk_msg('+12345', '+54321', transport_name=None)
+
+        with LogCatcher() as lc:
+            result_msg = yield mw.handle_inbound(msg, "dummy_connector")
+
+        self.assertEqual(
+            self.get_log(lc),
+            "Session length redis namespace cannot be None, skipping message")
```
Comment (hodgestar) [inline vumi/middleware/tests/test_session_length.py]: Any reason not to do
```python
self.assertEqual(
    lc.messages(),
    ["Session length ..."])
```
?

Comment (justinvdm) [inline vumi/middleware/tests/test_session_length.py]: Forgot about that, will do.

**Comment by hodgestar** [comment]
Otherwise looks good.

**Comment by justinvdm** [comment]
Ready for re-review.

**Comment by hodgestar** [comment]
:+1: as soon as Travis is happy.

**Pull request merged**